src/components/Form/Form.test.tsx

```diff
@@ -0,0 +1,101 @@
+import React from "react";
+import "@testing-library/jest-dom/extend-expect";
+import {
+  Form,
+  Input,
+  Button,
+  useBasisForm,
+} from "../..";
+
+import { render, screen, userEvent, act, fireEvent } from "../../utils/test";
+import { SubmitHandler } from "react-hook-form";
+
+
+interface SimpleFormValues {
+  testInput: string;
+}
+
+interface SimpleProps {
+  onSubmit: SubmitHandler<SimpleFormValues>;
+  validate?: any;
```

Comment:
Can basis provide a proper type for the validate function?

Reply:
Not really the validate is a very complex type and so even if basis were to provide it, you would need to pass into it 4 different parameters. In this case I circumvented it completely due to it being a test.

However in this case you would have to manually type the function to be:

`(val: string, props: InputProps) => string | string[] | null`.

If you incorrectly type it then it will give you a compile time error as it is internally checking those values against the one accepted by the `Component` passed in `as=` and the value derived from `name=`